lib/src/screens/add_card/select_merchant_screen.dart

```diff
@@ -53,10 +51,15 @@ class _SelectMerchantScreenState extends State<SelectMerchantScreen> {
             merchants.sort((a, b) => a.displayName.toLowerCase().compareTo(b.displayName.toLowerCase()));
 
             // Filter merchants based on search query
-            final filteredMerchants =
-            _searchQuery.isEmpty
+            final queryNormalized = removeDiacritics(_searchQuery.toLowerCase());
+
+            final filteredMerchants = _searchQuery.isEmpty
                 ? merchants
-                : merchants.where((merchant) => merchant.displayName.toLowerCase().contains(_searchQuery)).toList();
+                : merchants.where((merchant) {
+              final nameNormalized = removeDiacritics(merchant.displayName.toLowerCase());
+              return nameNormalized.contains(queryNormalized);
```

Comment:
variaveis desnecessárias :p 

pode ser só 

```suggestion
              return removeDiacritics(merchant.displayName.toLowerCase()).contains(queryNormalized);
```